frontend/src/pages/QuestPage.tsx

```diff
@@ -0,0 +1,246 @@
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import SideBar from '../components/SideBar';
+import Quest from '../components/Quest';
+import Input from '../components/Input';
+import Modal from '../components/Modal';
+import Button from '../components/Button';
+import QuestContainer from '../components/QuestContainer';
+import { DNDType } from '../utils/types';
+// DnD
+import {
+	DndContext,
+	DragEndEvent,
+	DragMoveEvent,
+	DragOverlay,
+	DragStartEvent,
+	KeyboardSensor,
+	PointerSensor,
+	UniqueIdentifier,
+	closestCorners,
+	useSensor,
+	useSensors,
+} from '@dnd-kit/core';
+import { SortableContext, sortableKeyboardCoordinates } from '@dnd-kit/sortable';
+import { onDragStart, onDragMove, onDragEnd } from '../utils/dragController';
+
+export default function QuestPage() {
+	const [containers, setContainers] = useState<DNDType[]>([
+		{
+			id: `container-${uuidv4()}`,
+			title: 'Task',
+			items: [
+			],
+		},
+		{
+			id: `container-${uuidv4()}`,
+			title: 'In Progress',
+			items: [
+			],
+		},
+	]);
+	const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+	const [currentContainerId, setCurrentContainerId] = useState<UniqueIdentifier>();
+	const [currentItemId, setCurrentItemId] = useState<UniqueIdentifier>();
+	const [itemName, setItemName] = useState('');
+	const [itemDescription, setItemDescription] = useState('');
+	const [showAddItemModal, setShowAddItemModal] = useState(false);
+	const [showEditItemModal, setShowEditItemModal] = useState(false);
+
+	const onAddItem = () => {
```

Comment:
In my opinion. I think this `onAddItem` and `onEditItem` function can merge into single function with receive parameters instead